File: CS50x/1. Apna College - CS50x/13.c
/*

> Date Created: June 15, 2023
> Author: Ishaan Rastogi
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working

> Practice Question 4: WAP to check if the number is divisible by 2 or not

/*

Topic: Operator Precedence Order
      
(NOT) ! > (* , / , %) > ( + , - ) > (<, <=, >, >=) > ( EQUAL TO, NOT EQUAL TO) (==, !=) > (AND) (&&) > (OR) (||) > (=) (Assignment operator)

*/


#include<stdio.h>

int main(void) {
    
    int a;
    printf("Enter a number:");
    scanf( "%d", &a );
    printf( "%d \n", a % 2 == 0 );

// even -> 1 ; odd -> 0 (Since 1 is for True & 0 for False)

}

/*

Ctrl + Shift + `

Terminal Code Loading...

gcc 13.c
./a.exe
 
*/

File: CS50x/6. Python/13.py
'''
> Date Created: December 25, 2023
> Author: Ishaan Rastogi
> Purpose: To rebuild mario blocks as we did in C
> Operating System: This is only for Windows OS, it may or may not work on other OS
> Program Status: 100% Working 
'''

'''

Topic- User Defined Functions

'''

#vertical block
n = int(input("Enter a number:")) #asking integer from user

for i in range(n):
    print("#")

#horizontal block
m = int(input("Enter a number:")) #asking integer from user

for j in range(m):
    print("?", end='') #end='' is used to remove end line character since '' denotes empty as there is nothing between the single inverted commas
print("") #to print a newline to tidy up the work, make sure the command is intended outside the for loop

#Shortest method using multiplication operator
p = int(input("Enter a number:")) #asking integer from user
print("?" * p)

'''
Ctrl + Shift + `
Terminal Code Loading...

python 13.py
'''